PowerEditor/visual.net/notepadPlus.vcxproj

```diff
@@ -130,7 +130,7 @@
       <Optimization>Disabled</Optimization>
       <FavorSizeOrSpeed>Neither</FavorSizeOrSpeed>
       <AdditionalIncludeDirectories>..\src\WinControls\AboutDlg;..\..\scintilla\include;..\..\lexilla\include;..\include;..\src\WinControls;..\src\WinControls\ImageListSet;..\src\WinControls\OpenSaveFileDialog;..\src\WinControls\SplitterContainer;..\src\WinControls\StaticDialog;..\src\WinControls\TabBar;..\src\WinControls\ToolBar;..\src\MISC\Process;..\src\ScintillaComponent;..\src\MISC;..\src\MISC\SysMsg;..\src\WinControls\StatusBar;..\src;..\src\WinControls\StaticDialog\RunDlg;..\src\tinyxml;..\src\WinControls\ColourPicker;..\src\Win32Explr;..\src\MISC\RegExt;..\src\WinControls\TrayIcon;..\src\WinControls\shortcut;..\src\WinControls\Grid;..\src\WinControls\ContextMenu;..\src\MISC\PluginsManager;..\src\WinControls\Preference;..\src\WinControls\WindowsDlg;..\src\WinControls\TaskList;..\src\WinControls\DockingWnd;..\src\WinControls\TreeView;..\src\WinControls\ToolTip;..\src\MISC\Exception;..\src\MISC\Common;..\src\tinyxml\tinyXmlA;..\src\WinControls\AnsiCharPanel;..\src\WinControls\ClipboardHistory;..\src\WinControls\FindCharsInRange;..\src\WinControls\VerticalFileSwitcher;..\src\WinControls\ProjectPanel;..\src\WinControls\DocumentMap;..\src\WinControls\FunctionList;..\src\uchardet;..\src\WinControls\FileBrowser;..\src\WinControls\ReadDirectoryChanges;..\src\MISC\md5;..\src\WinControls\PluginsAdmin;..\src\json;..\src\MISC\sha2;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
-      <PreprocessorDefinitions>WIN32;_WIN32_WINNT=0x0600;_WINDOWS;_USE_64BIT_TIME_T;TIXML_USE_STL;TIXMLA_USE_STL;_CRT_NONSTDC_NO_DEPRECATE;_CRT_SECURE_NO_WARNINGS;_CRT_NON_CONFORMING_SWPRINTFS=1;_CRT_NON_CONFORMING_WCSTOK;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <PreprocessorDefinitions>WIN32;_WIN32_WINNT=0x0600;_WINDOWS;_USE_64BIT_TIME_T;TIXML_USE_STL;TIXMLA_USE_STL;_CRT_NONSTDC_NO_DEPRECATE;_CRT_SECURE_NO_WARNINGS;_CRT_NON_CONFORMING_SWPRINTFS=1;_CRT_NON_CONFORMING_WCSTOK;_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING;%(PreprocessorDefinitions)</PreprocessorDefinitions>

```

Comment:
> Do we need _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING ? (Since there's no deprecation warning)

As I mentioned in PR description, it is needed to deal with deprecated warning. Just retry to rebuild you should get deprecated warning as below -


```
Build started...
1>------ Build started: Project: Notepad++, Configuration: Debug x64 ------
1>Buffer.cpp
1>X:\npp\PowerEditor\src\ScintillaComponent\Buffer.cpp(163,31): error C2220: the following warning is treated as an error
1>X:\npp\PowerEditor\src\ScintillaComponent\Buffer.cpp(163,31): warning C4996: 'std::codecvt_utf8_utf16<wchar_t,1114111,(std::codecvt_mode)0>': warning STL4017: std::wbuffer_convert, std::wstring_convert, and the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::codecvt_utf8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard doesn't provide equivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMultiByte() from <Windows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning.
...
```

> Why do we get warning cl : command line warning D9002: ignoring unknown option '/std:c++20' after removing _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING ?

I tried to compile using `MSBUILD` command. I get warning which eventually gets converted to error. I didn't get warning as you received. 

Command: `msbuild "X:\npp\PowerEditor\visual.net\notepadPlus.sln" /m`

Response:

```
Build FAILED.

       "X:\npp\PowerEditor\visual.net\notepadPlus.sln" (default target) (1)
        ->
       "X:\npp\PowerEditor\visual.net\notepadPlus.vcxproj.metaproj" (defaul
       t target) (2) ->
       "X:\npp\PowerEditor\visual.net\notepadPlus.vcxproj" (default target)
        (5) ->
       (ClCompile target) ->
         X:\npp\PowerEditor\src\ScintillaComponent\Buffer.cpp(163,31): warn
       ing C4996: 'std::codecvt_utf8_utf16<wchar_t,1114111,(std::codecvt_mode)0>': warning STL4017: std::wbuffer_conver
       t, std::wstring_convert, and the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt
       _utf16, and std::codecvt_utf8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated
       .) The C++ Standard doesn't provide equivalent non-deprecated functionality; consider using MultiByteToWideChar(
       ) and WideCharToMultiByte() from <Windows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_W
       ARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning. [F:\VS_Pro
       jects\OpenSourceContribution\gitGUI\nppMy\PowerEditor\visual.net\notepadPlus.vcxproj]
         X:\npp\PowerEditor\src\ScintillaComponent\Buffer.cpp(163,10): warn
       ing C4996: 'std::wstring_convert<std::codecvt_utf8_utf16<wchar_t,1114111,(std::codecvt_mode)0>,wchar_t,std::allo
       cator<wchar_t>,std::allocator<char>>': warning STL4017: std::wbuffer_convert, std::wstring_convert, and the <cod
       ecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::codecvt_utf8_utf16)
       are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard doesn't provide e
       quivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMultiByte() from <Win
       dows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECAT
       ION_WARNINGS to acknowledge that you have received this warning. [F:\VS_Projects\OpenSourceContribution\gitGUI\n
       ppMy\PowerEditor\visual.net\notepadPlus.vcxproj]
         X:\npp\PowerEditor\src\ScintillaComponent\Buffer.cpp(163,60): warn
       ing C4996: 'std::wstring_convert<std::codecvt_utf8_utf16<wchar_t,1114111,(std::codecvt_mode)0>,wchar_t,std::allo
       cator<wchar_t>,std::allocator<char>>::wstring_convert': warning STL4017: std::wbuffer_convert, std::wstring_conv
       ert, and the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::cod
       ecvt_utf8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard
       doesn't provide equivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMult
       iByte() from <Windows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING or _SILENCE_A
       LL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning. [F:\VS_Projects\OpenSourceCont
       ribution\gitGUI\nppMy\PowerEditor\visual.net\notepadPlus.vcxproj]
         X:\npp\PowerEditor\src\ScintillaComponent\Buffer.cpp(164,32): warn
       ing C4996: 'std::wstring_convert<std::codecvt_utf8_utf16<wchar_t,1114111,(std::codecvt_mode)0>,wchar_t,std::allo
       cator<wchar_t>,std::allocator<char>>::to_bytes': warning STL4017: std::wbuffer_convert, std::wstring_convert, an
       d the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::codecvt_ut
       f8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard doesn't
        provide equivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMultiByte()
        from <Windows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING or _SILENCE_ALL_CXX1
       7_DEPRECATION_WARNINGS to acknowledge that you have received this warning. [F:\VS_Projects\OpenSourceContributio
       n\gitGUI\nppMy\PowerEditor\visual.net\notepadPlus.vcxproj]
         X:\npp\PowerEditor\src\ScintillaComponent\Buffer.cpp(323,32): warn
       ing C4996: 'std::codecvt_utf8_utf16<wchar_t,1114111,(std::codecvt_mode)0>': warning STL4017: std::wbuffer_conver
       t, std::wstring_convert, and the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt
       _utf16, and std::codecvt_utf8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated
       .) The C++ Standard doesn't provide equivalent non-deprecated functionality; consider using MultiByteToWideChar(
       ) and WideCharToMultiByte() from <Windows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_W
       ARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning. [F:\VS_Pro
       jects\OpenSourceContribution\gitGUI\nppMy\PowerEditor\visual.net\notepadPlus.vcxproj]
         X:\npp\PowerEditor\src\ScintillaComponent\Buffer.cpp(323,11): warn
       ing C4996: 'std::wstring_convert<std::codecvt_utf8_utf16<wchar_t,1114111,(std::codecvt_mode)0>,wchar_t,std::allo
       cator<wchar_t>,std::allocator<char>>': warning STL4017: std::wbuffer_convert, std::wstring_convert, and the <cod
       ecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::codecvt_utf8_utf16)
       are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard doesn't provide e
       quivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMultiByte() from <Win
       dows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECAT
       ION_WARNINGS to acknowledge that you have received this warning. [F:\VS_Projects\OpenSourceContribution\gitGUI\n
       ppMy\PowerEditor\visual.net\notepadPlus.vcxproj]
         X:\npp\PowerEditor\src\ScintillaComponent\Buffer.cpp(323,61): warn
       ing C4996: 'std::wstring_convert<std::codecvt_utf8_utf16<wchar_t,1114111,(std::codecvt_mode)0>,wchar_t,std::allo
       cator<wchar_t>,std::allocator<char>>::wstring_convert': warning STL4017: std::wbuffer_convert, std::wstring_conv
       ert, and the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::cod
       ecvt_utf8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard
       doesn't provide equivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMult
       iByte() from <Windows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING or _SILENCE_A
       LL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning. [F:\VS_Projects\OpenSourceCont
       ribution\gitGUI\nppMy\PowerEditor\visual.net\notepadPlus.vcxproj]
         X:\npp\PowerEditor\src\ScintillaComponent\Buffer.cpp(324,33): warn
       ing C4996: 'std::wstring_convert<std::codecvt_utf8_utf16<wchar_t,1114111,(std::codecvt_mode)0>,wchar_t,std::allo
       cator<wchar_t>,std::allocator<char>>::to_bytes': warning STL4017: std::wbuffer_convert, std::wstring_convert, an
       d the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::codecvt_ut
       f8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard doesn't
        provide equivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMultiByte()
        from <Windows.h> instead. You can define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING or _SILENCE_ALL_CXX1
       7_DEPRECATION_WARNINGS to acknowledge that you have received this warning. [F:\VS_Projects\OpenSourceContributio
       n\gitGUI\nppMy\PowerEditor\visual.net\notepadPlus.vcxproj]


       "X:\npp\PowerEditor\visual.net\notepadPlus.sln" (default target) (1)
        ->
       "X:\npp\PowerEditor\visual.net\notepadPlus.vcxproj.metaproj" (defaul
       t target) (2) ->
       "X:\npp\PowerEditor\visual.net\notepadPlus.vcxproj" (default target)
        (5) ->
       (ClCompile target) ->
         X:\npp\PowerEditor\src\ScintillaComponent\Buffer.cpp(163,31): erro
       r C2220: the following warning is treated as an error [X:\npp\PowerE
       ditor\visual.net\notepadPlus.vcxproj]

    8 Warning(s)
    1 Error(s)

Time Elapsed 00:00:07.00
```

Reply:
@SinghRajenM 
Thank you for your reply.
Indeed, after update platform toolkits to v143, I can reproduce the warnings.